crates/chain-state/src/in_memory.rs

```diff
@@ -719,29 +719,37 @@ impl<N: NodePrimitives> BlockState<N> {
 }
 
 /// Represents an executed block stored in-memory.
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Clone, Debug)]
 pub struct ExecutedBlock<N: NodePrimitives = EthPrimitives> {
     /// Recovered Block
     pub recovered_block: Arc<RecoveredBlock<N::Block>>,
     /// Block's execution outcome.
     pub execution_output: Arc<ExecutionOutcome<N::Receipt>>,
-    /// Block's sorted hashed state.
-    pub hashed_state: Arc<HashedPostStateSorted>,
-    /// Sorted trie updates that result from calculating the state root for the block.
-    pub trie_updates: Arc<TrieUpdatesSorted>,
+    /// Deferred trie data produced by execution.
+    /// These are computed in a background task to unblock the validation hot path
+    pub trie_data: DeferredTrieData,
 }
 
 impl<N: NodePrimitives> Default for ExecutedBlock<N> {
     fn default() -> Self {
         Self {
             recovered_block: Default::default(),
             execution_output: Default::default(),
-            hashed_state: Default::default(),
-            trie_updates: Default::default(),
+            trie_data: DeferredTrieData::ready(ComputedTrieData::default()),
         }
     }
 }
 
+impl<N: NodePrimitives> PartialEq for ExecutedBlock<N> {
```

Comment:
In what situations is PartialEq/Eq for ExecutedBlock used? Could we just not support it?

Reply:
yeah I think if users need Eq, they should use the block hash